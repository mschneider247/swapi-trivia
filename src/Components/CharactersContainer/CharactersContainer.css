.charac-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
}

.favorite-que {
  margin-left: 0;
  color: white;
  font-size: 1.5rem;
}

.prompt {
  display: flex;
  align-items: center;
}

.heart {
  display: inline-block;
  background-image: url('../../images/swapi_heart_unclicked.svg');
  background-repeat: no-repeat;
  height: 43px;
  width: 53px;
}
