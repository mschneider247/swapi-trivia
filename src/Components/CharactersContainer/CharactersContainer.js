import React from 'react'
import './CharactersContainer.css';

const CharactersContainer = (props) => {
  console.log("these are CharactersContainer  props::", props)
  return (
    <section>
      <p>Characters Container</p>
    </section>
  )
}

export default CharactersContainer;