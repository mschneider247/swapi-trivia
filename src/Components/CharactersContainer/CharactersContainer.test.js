import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CharactersContainer from './CharactersContainer';

describe('CharactersContainer', () => {

  it('should match the snapshot with correct data', () => {
    const mockCurrentCharacters = [{
      films: "The Phantom Menace",
      homeworld: "Jabba Desilijic Tiure",
      name: "Jabba Desilijic Tiure",
      population: "7000000000",
      species: "Hutt",
      url: "https://swapi.co/api/people/16/"
    }, {
      films: "The Phantom Menace",
      homeworld: "Jabba Desilijic Tiure",
      name: "Jabba Desilijic Tiure",
      population: "7000000000",
      species: "Hutt",
      url: "https://swapi.co/api/people/16/"
    }]

    const mockUpdateFavoriteCharacters = jest.fn()

    const wrapper = shallow(
      <CharactersContainer
        characters={mockCurrentCharacters} updateFavoriteCharacters={mockUpdateFavoriteCharacters}
        favorites="true"
        checkFavoriteStatus="true"
      />
    );

    expect(wrapper).toMatchSnapshot()

  })

})
