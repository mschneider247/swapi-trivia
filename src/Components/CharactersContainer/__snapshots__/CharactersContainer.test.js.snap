// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersContainer should match the snapshot with correct data 1`] = `
<section
  className="charac-container"
>
  <Characters
    checkFavoriteStatus="true"
    films="The Phantom Menace"
    homeworld="Jabba Desilijic Tiure"
    id="16"
    isFavorite="true"
    key="16"
    name="Jabba Desilijic Tiure"
    population="7000000000"
    species="Hutt"
    updateFavoriteCharacters={[MockFunction]}
    url="https://swapi.co/api/people/16/"
  />
  <Characters
    checkFavoriteStatus="true"
    films="The Phantom Menace"
    homeworld="Jabba Desilijic Tiure"
    id="16"
    isFavorite="true"
    key="16"
    name="Jabba Desilijic Tiure"
    population="7000000000"
    species="Hutt"
    updateFavoriteCharacters={[MockFunction]}
    url="https://swapi.co/api/people/16/"
  />
</section>
`;
